lambdas/ec2/reaper.py

```diff
@@ -64,112 +64,104 @@ def wait_for_tags(ec2_instance, wait_time):
     for the 'lifetime' key, parses it, and sets the 'termination_date' on the
     instance. The 'termination_date' can be set directly on the instance, bypassing
     the steps to parse the lifetime key and allowing this to return.
+
+    This returns the termination_date value if successful; otherwise, it returns
+    None.
     """
     start = timenow_with_utc()
     timeout = start + datetime.timedelta(minutes=wait_time)
 
     while timenow_with_utc() < timeout:
         ec2_instance.load()
-        if get_tag(ec2_instance, 'termination_date'):
+        termination_date = get_tag(ec2_instance, 'termination_date')
+        if termination_date:
             print("'termination_date' tag found!")
-            return
-        if get_tag(ec2_instance, 'lifetime') is None:
+            return termination_date
+        lifetime = get_tag(ec2_instance,'lifetime')
+        if not lifetime:
             print("No 'lifetime' tag found; sleeping for 15s")
             time.sleep(15)
             continue
-        print('lifetime tag found')
-        lifetime = get_tag(ec2_instance,'lifetime')
 
-        try:
-            lifetime_delta = calculate_lifetime_delta(lifetime)
-        except ValueError as e:
-            terminate_and_raise_message(ec2_instance,
-                                        'Unable to parse the lifetime value',
-                                        exception=e)
-
-        termination_date = start + lifetime_delta
+        print('lifetime tag found')
+        lifetime_match = validate_lifetime_value(lifetime)
+        if not lifetime_match:
+            terminate_instance(ec2_instance, 'Invalid lifetime value supplied')
+            return
+        lifetime_delta = calculate_lifetime_delta(lifetime_match)
+        future_termination_date = start + lifetime_delta
         ec2_instance.create_tags(
                 Tags=[
                     {
                         'Key': 'termination_date',
-                        'Value': termination_date.isoformat()
+                        'Value': future_termination_date.isoformat()
                     }
                 ]
             )
 
     # If the above while condition does not return after finding a termination_date,
     # terminate the instance and raise an exception.
-    terminate_and_raise_message(ec2_instance,
-                                'No termination_date found within {0} minutes of creation'.format(wait_time))
+    terminate_instance(ec2_instance,
+                       'No termination_date found within {0} minutes of creation'.format(wait_time))
+    return
 
 
-def terminate_and_raise_message(ec2_instance, message, exception=None, live_mode=LIVE_MODE):
+def terminate_instance(ec2_instance, message='Terminating EC2 instance'):
```

Comment:
@tvpartytonight it seems that providing a default `message` here will allow people to call this function without providing more context, which could be needed in further debugging / maintenance. It looks like all of your references to this function do provide messages, so why don't we not provide a default & fail if people don't provide one, making this function much more informative & easier to work with in the future?